async-query-core/src/main/java/org/opensearch/sql/spark/utils/SQLQueryUtils.java

```diff
@@ -77,32 +83,73 @@ public static boolean isFlintExtensionQuery(String sqlQuery) {
     }
   }
 
-  public static List<String> validateSparkSqlQuery(String sqlQuery) {
-    SparkSqlValidatorVisitor sparkSqlValidatorVisitor = new SparkSqlValidatorVisitor();
-    SqlBaseParser sqlBaseParser =
-        new SqlBaseParser(
-            new CommonTokenStream(new SqlBaseLexer(new CaseInsensitiveCharStream(sqlQuery))));
-    sqlBaseParser.addErrorListener(new SyntaxAnalysisErrorListener());
+  public static List<String> validateSparkSqlQuery(DataSource datasource, String sqlQuery) {
     try {
-      SqlBaseParser.StatementContext statement = sqlBaseParser.statement();
-      sparkSqlValidatorVisitor.visit(statement);
-      return sparkSqlValidatorVisitor.getValidationErrors();
-    } catch (SyntaxCheckException syntaxCheckException) {
+      SqlBaseParser sqlBaseParser =
+          new SqlBaseParser(
+              new CommonTokenStream(new SqlBaseLexer(new CaseInsensitiveCharStream(sqlQuery))));
+      sqlBaseParser.addErrorListener(new SyntaxAnalysisErrorListener());
+      SqlBaseValidatorVisitor sqlParserBaseVisitor = getSparkSqlValidatorVisitor(datasource);
+      StatementContext statement = sqlBaseParser.statement();
+      sqlParserBaseVisitor.visit(statement);
+      return sqlParserBaseVisitor.getValidationErrors();
+    } catch (SyntaxCheckException e) {
+      logger.error(
+          String.format(
+              "Failed to parse sql statement context while validating sql query %s", sqlQuery),
+          e);
       return Collections.emptyList();
     }
   }
 
-  private static class SparkSqlValidatorVisitor extends SqlBaseParserBaseVisitor<Void> {
+  private SqlBaseValidatorVisitor getSparkSqlValidatorVisitor(DataSource datasource) {
+    if (datasource != null
+        && datasource.getConnectorType() != null
+        && datasource.getConnectorType().equals(DataSourceType.SECURITY_LAKE)) {
+      return new SparkSqlSecurityLakeValidatorVisitor();
+    } else {
+      return new SparkSqlValidatorVisitor();
+    }
+  }
 
-    @Getter private final List<String> validationErrors = new ArrayList<>();
+  /**
+   * A generic validator impl for Spark Sql Queries that supports accumulating validation errors on
+   * visiting sql statement
```

Comment:
nit: this sounds a bit weird to me. should it say this is a base class instead of generic validator impl?

Reply:
fixed